ast/parser_ext.go

```diff
@@ -686,6 +715,21 @@ func setRuleModule(rule *Rule, module *Module) {
 	}
 }
 
+func setTermsJSONOptions(bodyTerms interface{}, jsonOptions *JSONOptions) {
+	switch terms := bodyTerms.(type) {
+	case []*Term:
+		for _, term := range terms {
+			term.setJSONOptions(*jsonOptions)
+		}
+	case *SomeDecl:
+		terms.setJSONOptions(*jsonOptions)
+	case *Every:
+		terms.setJSONOptions(*jsonOptions)
+	case *Term:
+		terms.setJSONOptions(*jsonOptions)
```

Comment:
Would this be the default case?

Reply:
Thanks for the clarification. So something like below. Also we don't need a default case, as it would have been caught somewhere before we even reach this place in the code flow, correct?

```go
case *SomeDecl, *Every, *Term:
        terms.setJSONOptions(*jsonOptions)
```

Reply:
It should be caught by the `panic` in [NewExpr](https://github.com/open-policy-agent/opa/blob/main/ast/policy.go#L1151) when the body is created in the `Parse*From*` functions. I will substitute the multiple cases with the following code:

```go
case customJSON:
        terms.setJSONOptions(*jsonOptions)
```

to avoid `terms` having the `interface{}` type.

I will also reflect this change in the test cases in `parser_text.go`